### PR TITLE
Code review for features branch

### PR DESCRIPTION
**Pull request by javitocor**
In this project we have built a website where you can signup, login and add posts:
- We have used devise gem for authenticate users and create/destroy sessions. 
- We have created a posts model with a foreign key in the database to 'users' table
- We have created a post controller with index, new and create actions
- We have created the associations between models
- We have created the views for the posts controller actions, and a partial form
- We have added logic to our posts views so only a signed user can see the name of the user who created the post
- Only logged users can create posts

The project has passed rubocop test

### CODE REVIEW

**Comment by javitocor** [comment]
We have fixed all the rubocop errors, see last commit, and at our end we see 'no offenses' after testing the code with rubocop. The errors remain here, in the pull request, but we cannot fix them because in our end everything looks good.

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[new create]
```
Comment (shshamim63) [inline app/controllers/posts_controller.rb]: You are using the helper method provided by 'devise'. But based on the instruction below,
screenshot 
What I can get is you have to generate a necessary helper method that will handle the task. I tried something like
```ruby
# created a method in app/controllers/application_controller.rb
def signed_in_only!
    redirect_to  new_user_session_path unless user_signed_in?
end
```
After that, replaced authenticate_user! with the method I created. Now the biggest question if devise provides me a sweet and short option to handle it, why I have to do it? Well, to ensure that you know if your ever have to use them inside other methods, you know that it is possible.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :traffic_light: Required Changes :tractor: 
___
Hi @javitocor and @tirthajyoti-ghosh :wave: 
Awesome job :clap: Repository contains a well descriptive readme, code looks neat and clean, PR contains all the info to highlight a review about the task you have covered. I really liked the way you have commented on the PR based on the issue you have faced. Amazing job :raised_hands: But there are some issues that are blocking me. Let's get to them.
___
### :spades: Mandatory Changes :spades: 
___
1. Based on the Odin project instruction I think you have to add a helper method. Check out the comment I have added.
2. The way you have tried to fix the linter error is not the actual way to solve it. The error is related to carriage return. To fix the error you have to add a simple line in your rubocop.yml file. that is 
```yml
Style/EndOfLine:
  EnforcedStyle: crlf
```
I think It will fix it. Check out these [link1](https://stackoverflow.com/questions/45157029/carriage-return-character-missing-style-endofline-in-ruby-on-rails) [link2](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/EndOfLine)
___
#### Please fix the above-mentioned issue and send another review request(feel free to knock me for any concern).
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

**Comment by tirthajyoti-ghosh** [comment]
Thanks, @shshamim63 for an awesome and descriptive review!
Changes are done after review:
- Fixed carriage return rubocop errors.
- Added custom helper method (`enforce_log_in`) in `app/controllers/application_controller.rb` to enforce login before creating new posts. Previously devise helper method `authenticate_user!` was used.

**Review by mosesogwo — APPROVED** [review]
**Great! Project is approved!**

_**Good luck in your next project.:stars:**_

### Status: Approved:heavy_check_mark:

**Pull request merged**